include/mbedtls/check_config.h

```diff
@@ -333,13 +333,17 @@
 #endif
 #undef MBEDTLS_HAS_MEMSAN
 
-#if defined(MBEDTLS_CCM_C) && (                                        \
-    !defined(MBEDTLS_AES_C) && !defined(MBEDTLS_CAMELLIA_C) && !defined(MBEDTLS_ARIA_C) )
+#if defined(MBEDTLS_CCM_C) && \
+    !(defined(MBEDTLS_BLOCK_CIPHER_CAN_AES) || defined(MBEDTLS_AES_C) || \
```

Comment:
That's not quite correct now: `CCM_C` can't benefit from AES support in `BLOCK_CIPHER_C` when `CIPHER_C` is defined, but I think that's something we should change anyway, and then this would become correct.

Reply:
As far as I see we have 2 options:
- if `CIPHER_C` is defined, then CCM/GCM can do AES (through CIPHER) as long as `AES_C` if defined
- if `CIPHER_C` is not defined, then CCM/GCM can do AES (though BLOCK_CIPHER) as long as `MBEDTLS_BLOCK_CIPHER_CAN_AES` is defined. Here `MBEDTLS_BLOCK_CIPHER_CAN_AES` strictly depends on `BLOCK_CIPHER_C`, of course,

So I think that the above line should be:
`defined(MBEDTLS_BLOCK_CIPHER_CAN_AES) || (defined(MBEDTLS_CIPHER_C) && defined(MBEDTLS_AES_C))`. Isn't it?

[I wrote about AES here, but the same holds for ARIA and Camellia.]

> I think that's something we should change anyway, and then this would become correct.

Is this related to [this comment](https://github.com/Mbed-TLS/mbedtls/pull/8632#discussion_r1425220251)?

Reply:
Yes, that was related to that comment, and I think it's correct now, so I'm marking this conversation as resolved.